spec/requests/whitehall_migration_spec.rb

```diff
@@ -0,0 +1,46 @@
+RSpec.describe "Whitehall Migration" do
+  let(:whitehall_migration) { create(:whitehall_migration) }
+  let(:document) { create(:whitehall_migration_document_import, whitehall_migration: whitehall_migration) }
+
+  it_behaves_like "requests that return status",
+                  "when a user without debug permissions looks at a whitehall migration",
+                  status: :forbidden,
+                  routes: { whitehall_migration_path: %i[get],
+                          whitehall_migration_documents_path: %i[get] } do
+    before { login_as(create(:user)) }
+
+    let(:route_params) { [whitehall_migration] }
+  end
+
+  it_behaves_like "requests that return status",
+                  "when a user without debug permissions looks at a whitehall migration document",
+                  status: :forbidden,
+                  routes: { whitehall_migration_document_path: %i[get] } do
+    before { login_as(create(:user)) }
+
+    let(:route_params) do
+      [whitehall_migration, document]
+    end
+  end
+
+  it_behaves_like "requests that return status",
+                  "when a user with debug permissions looks at a whitehall migration",
+                  status: :success,
+                  routes: { whitehall_migration_path: %i[get],
+                          whitehall_migration_documents_path: %i[get] } do
+    before { login_as(create(:user, permissions: [User::DEBUG_PERMISSION])) }
+
+    let(:route_params) { [whitehall_migration] }
+  end
+
+  it_behaves_like "requests that return status",
+                  "when a user with debug permissions looks at a whitehall migration document",
+                  status: :success,
+                  routes: { whitehall_migration_document_path: %i[get] } do
+    before { login_as(create(:user, permissions: [User::DEBUG_PERMISSION])) }
+
+    let(:route_params) do
+      [whitehall_migration, document]
+    end
+  end
```

Comment:
We don't tend to have any precedent for using these `it_behaves_like "requests that return status",` for successful scenarios. 

It'd be more consistent with the app if we had regular request tests for success scenario. We also have precedent of asserting on status ok rather than success - for example: https://github.com/alphagov/content-publisher/blob/0ba258d25eba53016416e41e28c347dee96433e0/spec/requests/access_limit_spec.rb#L30-L34